### PR TITLE
Simplified reverse in lease majority

### PR DESCRIPTION
**Pull request by octonato**
Somehow I forgot to sent this PR. This was triggered while doing some code review in #29737  (discussed [here](https://github.com/akka/akka/pull/29737#discussion_r532433436))

### CODE REVIEW

File: akka-cluster/src/main/scala/akka/cluster/sbr/DowningStrategy.scala [hunk 1]
```diff
@@ -317,17 +317,10 @@ import akka.coordination.lease.scaladsl.Lease
     unreachableMembers.forall(m => m.status == MemberStatus.Down || m.status == MemberStatus.Exiting)
   }
 
-  def reverseDecision(decision: Decision): Decision = {
+  def reverseDecision(decision: AcquireLeaseDecision): Decision = {
```
Comment (octonato) [inline akka-cluster/src/main/scala/akka/cluster/sbr/DowningStrategy.scala:320]: That's the main change. That method is only called when `Decision` is already narrow down to `AcquireLeaseDecision`. We don't need pattern match on the others.

All the reversing logic is useless except for `AcquireLeaseAndDownUnreachable` and `AcquireLeaseAndDownIndirectlyConnected`.

File: akka-cluster/src/test/scala/akka/cluster/sbr/SplitBrainResolverSpec.scala [hunk 2]
```diff
@@ -846,15 +846,15 @@ class SplitBrainResolverSpec
       val decision1 = strategy1.decide()
       decision1 should ===(AcquireLeaseAndDownUnreachable(Duration.Zero))
       strategy1.nodesToDown(decision1) should ===(side2Nodes)
-      val reverseDecision1 = strategy1.reverseDecision(decision1)
+      val reverseDecision1 = strategy1.reverseDecision(decision1.asInstanceOf[AcquireLeaseDecision])
```
Comment (octonato) [inline akka-cluster/src/test/scala/akka/cluster/sbr/SplitBrainResolverSpec.scala:849]: Annoying to have to add the case, but at least the prod code is well-typed now and we force the tests to follow the types.

File: akka-cluster/src/main/mima-filters/2.6.13.backwards.excludes/simplified-reverse-logic-in-lease-majority.excludes [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# #30155 simplified reverse logic in lease majority
+ProblemFilters.exclude[IncompatibleMethTypeProblem]("akka.cluster.sbr.SplitBrainResolverBase.waitingForLease")
+ProblemFilters.exclude[IncompatibleMethTypeProblem]("akka.cluster.sbr.DowningStrategy.reverseDecision")
```
Comment (octonato) [inline akka-cluster/src/main/mima-filters/2.6.13.backwards.excludes/simplified-reverse-logic-in-lease-majority.excludes:3]: ok to exclude, both private / internal api

File: akka-cluster/src/main/scala/akka/cluster/sbr/DowningStrategy.scala (hunk 1 above)
Comment (johanandren) [reply akka-cluster/src/main/scala/akka/cluster/sbr/DowningStrategy.scala:320]: Ah, I didn't get that when you commented on it, nice.

**Review by johanandren — APPROVED** [review]

**Review by chbatey — APPROVED** [review]

**Pull request merged**